proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+
+This proposal only accounts for the **majority of packages on NuGet.org**. There are many unique TFMs in the .NET ecosystem, but for the sake of creating a concept that will scale with the future of .NET, many unsupported & low usage TFMs are not considered. Thus this proposal only focuses on the [most common target frameworks](https://docs.microsoft.com/dotnet/standard/frameworks#latest-versions) & omits considering TFMs with fewer than 5,000 packages.
+
+#### Generations of .NET
+
+There are many generations of .NET. For the sake of this proposal, we will focus on defining the three major generations:
+
+- .NET Framework:
+  - `net, net11, net20, net35, net40, net403, net45, net451, net452, net46, net461, net462, net463, net47, net471, net472, net48`
+- .NET Core:
+  - `netcoreapp1.0, netcoreapp1.1, netcoreapp2.0, netcoreapp2.1, netcoreapp2.2, netcoreapp3.0, netcoreapp3.1`
+- .NET:
+  - `net5.0, net6.0`
+- Any:
+  - .NET Framework, .NET Core, .NET
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/GenerationsOfNET.png)
+
+#### Supported Platforms
+
+With .NET Core introducing support for cross-platform scenarios, we will define the following as supported platforms:
+
+- .NET Standard:
+  - `netstandard, netstandard1.0, netstandard1.1, netstandard1.2, netstandard1.3, netstandard1.4, netstandard1.5, netstandard1.6, netstandard2.0, netstandard2.1`
+- Windows:
+  - .NET Framework
+  - .NET Core
+  - .NET 5+
+    - `net5.0-windows, net6.0-windows`
+  - .NET Standard
+- macOS:
+  - .NET Core
+  - .NET 5+
+    - `net6.0-macos`
+  - .NET Standard
+- Linux:
+  - .NET Core
+  - .NET 5+
+  - .NET Standard
+- Android:
+  - .NET 6+
+    - `net6.0-android`
+  - .NET Standard
+- iOS:
+  - .NET 6+
+    - `net6.0-ios`
+  - .NET Standard
+- macCatalyst:
+  - .NET 6+
+    - `net6.0-maccatalyst`
+  - .NET Standard
+- tvOS:
+  - .NET 6+
+    - `net6.0-tvos`
+  - .NET Standard
+- Tizen:
+  - .NET 6+
+    - `net6.0-tizen`
+  - .NET Standard
+
+**Note:** .NET Standard is considered a special case that supports majority of these platforms depending on the [version of .NET standard](https://docs.microsoft.com/dotnet/standard/net-standard#net-implementation-support). If the framework precedence of a target framework supports .NET Standard, it's respective versions supported will be implicitly included.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/SupportedPlatforms.png)
+
+#### Version Numbers
+
+There are many unique version numbers for all generations of .NET & supported platforms. The filtering experience should be smart enough to create a version list based on the current selection. Here are a few scenarios:
+
+- User selects `Any` for the .NET generation.
+  - Version filter populates all known versions for .NET Framework, .NET Core, .NET, and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
```

Comment:
I have concerns about showing .NET Framework packages as compatible here, especially later versions (such as .NET 4.8).

These are not fully compatible. Should we differentiate between the 2?

If we won't we should call it out explicitly.

Reply:
In this scenario, it's just a filter for "Any". 

If a user selected `.NET Framework` & a specific version of framework like 4.5, they should only see `.NET Standard 1.0-1.1` compatible assets.

i.e. net45+ https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support

Reply:
makes sense. Resolving.